src/genotyper.cc

```diff
@@ -347,14 +335,96 @@ static Status translate_genotypes(const genotyper_config& cfg, const unified_sit
     return Status::OK();
 }
 
+/// Given a set of records overlapping a site, we find the
+/// "representative record(s)" for a given site. The representative
+/// record(s) found are returned via the ans vector. This function also
+/// modifies 2 genotyper substrates: failed_ref_depth and genotypes.
+/// Failed_ref_depth tracks the ref-records for a given sample has failed
+/// the minimal depth requirements (if any).
+/// Modifications to genotypes vector is explained below
+/// The representative record(s) are expected to be passed into translate_genotypes for genotyping
+/// =====================================================
+/// Pre conditions:
+/// ans should be an empty vector when passed into the function, otherwise an invalid status will be raised
+/// ======================================================
+/// Expected behaviors:
+/// No records given
+///      Empty ans is returned, no modification to genotypes and failed_ref_depth vectors
+/// Records do not span the entire range of site
+///      Empty ans is returned, failed_ref_depth updated accordingly, no change to genotypes
+/// Records span entire range, and consist of all gvcf records
+///      ans contains the first gvcf record passed, failed_ref_depth updated accordingly, no change to genotypes
+/// Records span entire range, and consist of exactly 1 vcf variant record
+///      ans contains the 1 vcf variant record, failed_ref_depth updated accordingly, no change to genotypes
+/// Records span entire range, and consists of >1 vcf variant record
+///      Empty ans is returned, failed_ref_depth updated accordingly, genotypes for all samples in this dataset are updated to be lost calls with RNC=LostAllele
+
```

Comment:
As I understand it `find_rep_record` is meant to "return" either zero or one records, currently in a `vector<shared_ptr<bcf1_t>> ans` of length either zero or one. You could simplify this to `shared_ptr<bcf1_t>& ans` which is set on exit to either `nullptr` or filled. (Unless, do you envision future cases where you'll want to produce more than one record?)